Optional generic parameter can't use generic type in local function

**Version Used**: 
VS 2017 RTM

**Steps to Reproduce**:
Add the following as a local function:

```csharp
T GetField<T>(string name, T @default = default(T))
{
  return @default;
}
```

This works as a normal method, but not as a local function.

**Expected Behavior**:

Compiles and can use the local function.

**Actual Behavior**:

```
error CS0246: The type or namespace name 'T' could not be found (are you missing a using directive or an assembly reference?)
error CS1750: A value of type 'T' cannot be used as a default parameter because there are no standard conversions to type 'T'
```

--- a/src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenLocalFunctionTests.cs
+++ b/src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenLocalFunctionTests.cs
@@ -31,6 +31,25 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests.CodeGen
     public class CodeGenLocalFunctionTests : CSharpTestBase
     {
         [Fact]
+        [WorkItem(17719, "https://github.com/dotnet/roslyn/issues/17719")]
+        public void Repro17719()
+        {
+            var comp = CompileAndVerify(@"
+using System;
+class C
+{
+    public static void Main()
+    {
+        T GetField<T>(string name, T @default = default(T))
+        {
+          return @default;
+        }
+        Console.WriteLine(GetField<int>(string.Empty));
+    }
+}", expectedOutput: "0");
+        }
+
+        [Fact]
         [WorkItem(17890, "https://github.com/dotnet/roslyn/issues/17890")]
         public void Repro17890()
         {

